### PR TITLE
Bhr 31 

### PR DESCRIPTION
**Pull request by RobertHebel**
Create action and reducer for fetching book data by ISBN number (call to backend)

### CODE REVIEW

File: backend/src/book/controller.js [hunk 1]
```diff
@@ -38,7 +38,10 @@ const parseBookdata = data => ({
   },
 });
 
-export const fetchBookDataByIsbn = (req, res, next) => booksApiClient.fetchBookByIsbn(req.isbn)
-  .then(parseBookdata)
-  .then(book => res.json(book))
-  .catch(next);
+export const fetchBookDataByIsbn = (req, res, next) => {
+  console.log('fetchingData');
```
Comment (adam-golab) [inline backend/src/book/controller.js]: Remove this `console.log`

Comment (adam-golab) [inline backend/src/book/controller.js]: Remove this `console.log`

File: backend/src/index.js [hunk 2]
```diff
@@ -3,26 +3,29 @@ import express from 'express';
...
+
 import cors from 'cors';
 
 const app = express();
 
 app.use(cors());
 
 app.get('/', function(req, res) {
-  res.send('Hello World!');
+    res.send('Hello World!');
```
Comment (adam-golab) [inline backend/src/index.js]: Why did you add this unnecessary 2 spaces?

File: backend/src/index.js [hunk 3]
```diff
@@ -3,26 +3,29 @@ import express from 'express';
...
-  .then(() => {
-    app.db = db;
-    app.listen(process.env.PORT, function() {
-      console.log(`App listening on port ${process.env.PORT}!`);
-    });
-  })
-  .catch(err => {
-    console.error('Unable to connect to the database:', err);
-  });
+    .then(() => {
```
Comment (adam-golab) [inline backend/src/index.js]: To large indentation.

File: frontend/src/containers/addBook/BookDescription.js [hunk 4]
```diff
@@ -1,88 +1,134 @@
...
-      label="Language"
-      options={['English', 'Polish', 'Spanish']}
-    />
-    <Field
-      name="description"
-      component={ TextArea }
-      type="textArea"
-      label="Description"
-      placeholder="In this book you will learn: - Everything you need to know about Python:
+class BookDescription extends PureComponent {
```
Comment (adam-golab) [inline frontend/src/containers/addBook/BookDescription.js]: Why PureComponent?

File: frontend/src/containers/addBook/SearchBook.js [hunk 5]
```diff
@@ -27,10 +29,9 @@ class SearchBook extends Component {
       this.setState({
```
Comment (adam-golab) [inline frontend/src/containers/addBook/SearchBook.js]: In `constructor` you can just set state via `this.state = { yourInitialState }`. And you forgot about `super(props)` here

Comment (RobertHebel) [inline frontend/src/containers/addBook/SearchBook.js]: I think that I just did it ;)

File: backend/src/index.js (hunk 2 above)
Comment (RobertHebel) [inline backend/src/index.js]: #justWebstormThings

File: frontend/src/containers/addBook/BookDescription.js [hunk 6]
```diff
@@ -1,88 +1,134 @@
...
+      description: PropTypes.string,
+      language: PropTypes.string,
+      publisher: PropTypes.string,
+      year: PropTypes.string,
+      pages: PropTypes.number,
+    }),
+    dispatch: PropTypes.func.isRequired,
+  };
+
+  // It will be changed in the future
```
Comment (adam-golab) [inline frontend/src/containers/addBook/BookDescription.js]: Add `TODO:` to comment

File: frontend/src/reducers/fetchBookReducer.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import { FETCH_REQUEST, FETCH_SUCCESS, FETCH_FAILURE } from '../constants/actionTypes';
+import initialState from './initialState';
+
+export default function(state = initialState.bookDetails, action) {
+  switch (action.type) {
+    case FETCH_REQUEST:
+      return {
+        ...state,
+        isbn: 0,
+        data: {},
```
Comment (adam-golab) [inline frontend/src/reducers/fetchBookReducer.js]: data: null

File: frontend/src/reducers/index.js [hunk 8]
```diff
@@ -9,6 +10,7 @@ const rootReducer = combineReducers({
   routing: routerReducer,
   form: formReducer,
   books: booksReducer,
+  bookDetails: fetchBookReducer,
```
Comment (adam-golab) [inline frontend/src/reducers/index.js]: Think about better name ;)

File: frontend/src/reducers/initialState.js [hunk 9]
```diff
@@ -14,4 +14,14 @@ export default {
     data: {},
     isTokenConsumed: false,
   },
+  bookDetails: {
+    isbn: 0,
+    data: {},
```
Comment (adam-golab) [inline frontend/src/reducers/initialState.js]: `null`

File: frontend/src/reducers/initialState.js [hunk 10]
```diff
@@ -14,4 +14,14 @@ export default {
     data: {},
```
Comment (adam-golab) [inline frontend/src/reducers/initialState.js]: `null`
